src/librustc/ty/layout.rs

```diff
@@ -1513,6 +1531,59 @@ impl<'a, 'gcx, 'tcx> Layout {
             }
         }
     }
+
+    pub fn field_offset(&self,
```

Comment:
We'll hopefully eventually remove `variant_index` by adding variant types, so I'm not sure how much I care about it.